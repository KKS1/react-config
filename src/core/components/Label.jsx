import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
