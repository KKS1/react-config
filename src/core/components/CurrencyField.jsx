import React, { Component } from "react";
import MaskedInput from "./MaskedInput";
import Label from "./Label";

export default class CurrencyField extends Component {
  static defaultProps = {
    label: "Label",
    placeholder: "",
    name: "currencyFormField",
    onChange: () => {}
  };

  changeHandler = e => {
    this.props.onChange(e.target.value);
  };

  render() {
    const {
      label,
      placeholder,
      name,
      value,
      id,
      options,
      disabled
    } = this.props;
    const { maskconfig } = options || {};

    return (
      <div className="c-label-field">
        <Label {...this.props} />
        <div className="d-flex">
          <i className="far fa-dollar-sign c-label-field__icon c-label-field__icon--prepend" />
          <MaskedInput
            id={id}
            className="c-label-field__input c-label-field__input--icon-prefixed"
            type="text"
            inputMode="numeric"
            placeholder={placeholder}
            name={name}
            disabled={disabled}
            value={value == null ? 0 : value}
            maskconfig={maskconfig}
            onChange={this.changeHandler}
          />
        </div>
      </div>
    );
  }
}
