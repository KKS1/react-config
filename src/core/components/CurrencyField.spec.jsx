import React, { Component } from "react";
import Label from "./Label";
import CurrencyField from "./CurrencyField";

describe("CurrencyField", () => {
  test("render_CurrencyField", () => {
    const wrapper = shallow(<CurrencyField />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("attach_props", () => {
    const props = {
      id: "formOne_labelOne",
      value: 100,
      options: {
        maskconfig: {
          min: 0,
          max: 100000,
          digits: 2
        }
      }
    };
    const wrapper = shallow(<CurrencyField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("call_onChange_prop", () => {
    const changeHandler = jest.fn();

    const props = {
      id: "formOne_labelOne",
      value: 100,
      onChange: changeHandler
    };
    const wrapper = shallow(<CurrencyField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();

    instance.changeHandler({ target: { value: 200 } });

    expect(changeHandler).toHaveBeenCalledWith(200);
  });
});
