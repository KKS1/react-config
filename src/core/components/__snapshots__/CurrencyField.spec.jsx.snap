// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyField attach_props 1`] = `
CurrencyField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "currencyFormField",
    "onChange": [Function],
    "options": Object {
      "maskconfig": Object {
        "digits": 2,
        "max": 100000,
        "min": 0,
      },
    },
    "placeholder": "",
    "value": 100,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <CurrencyField
        id="formOne_labelOne"
        label="Label"
        name="currencyFormField"
        onChange={[Function]}
        options={
          Object {
            "maskconfig": Object {
              "digits": 2,
              "max": 100000,
              "min": 0,
            },
          }
        }
        placeholder=""
        value={100}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="currencyFormField"
          onChange={[Function]}
          options={
            Object {
              "maskconfig": Object {
                "digits": 2,
                "max": 100000,
                "min": 0,
              },
            }
          }
          placeholder=""
          value={100}
        />
        <div
          className="d-flex"
        >
          <i
            className="far fa-dollar-sign c-label-field__icon c-label-field__icon--prepend"
          />
          <MaskedInput
            className="c-label-field__input c-label-field__input--icon-prefixed"
            id="formOne_labelOne"
            inputMode="numeric"
            maskconfig={
              Object {
                "digits": 2,
                "max": 100000,
                "min": 0,
              }
            }
            name="currencyFormField"
            onChange={[Function]}
            placeholder=""
            type="text"
            value={100}
          />
        </div>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`CurrencyField call_onChange_prop 1`] = `
CurrencyField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "currencyFormField",
    "onChange": [MockFunction],
    "placeholder": "",
    "value": 100,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <CurrencyField
        id="formOne_labelOne"
        label="Label"
        name="currencyFormField"
        onChange={[MockFunction]}
        placeholder=""
        value={100}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="currencyFormField"
          onChange={[MockFunction]}
          placeholder=""
          value={100}
        />
        <div
          className="d-flex"
        >
          <i
            className="far fa-dollar-sign c-label-field__icon c-label-field__icon--prepend"
          />
          <MaskedInput
            className="c-label-field__input c-label-field__input--icon-prefixed"
            id="formOne_labelOne"
            inputMode="numeric"
            maskconfig={
              Object {
                "alias": "numeric",
                "autoGroup": true,
                "digits": 0,
                "digitsOptional": true,
                "groupSeparator": ",",
                "min": 0,
                "rightAlign": false,
              }
            }
            name="currencyFormField"
            onChange={[Function]}
            placeholder=""
            type="text"
            value={100}
          />
        </div>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`CurrencyField render_CurrencyField 1`] = `
CurrencyField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "label": "Label",
    "name": "currencyFormField",
    "onChange": [Function],
    "placeholder": "",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <CurrencyField
        label="Label"
        name="currencyFormField"
        onChange={[Function]}
        placeholder=""
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          label="Label"
          name="currencyFormField"
          onChange={[Function]}
          placeholder=""
        />
        <div
          className="d-flex"
        >
          <i
            className="far fa-dollar-sign c-label-field__icon c-label-field__icon--prepend"
          />
          <MaskedInput
            className="c-label-field__input c-label-field__input--icon-prefixed"
            inputMode="numeric"
            maskconfig={
              Object {
                "alias": "numeric",
                "autoGroup": true,
                "digits": 0,
                "digitsOptional": true,
                "groupSeparator": ",",
                "min": 0,
                "rightAlign": false,
              }
            }
            name="currencyFormField"
            onChange={[Function]}
            placeholder=""
            type="text"
            value={0}
          />
        </div>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
